addons/web_editor/tools.py

```diff
@@ -0,0 +1,134 @@
+
+import base64
+import json
+import re
+import requests
+
+from odoo import _
+from odoo.tools import image_process
+from odoo.modules.module import get_module_resource
+
+
+# To detect if we have a valid URL or not
+valid_url_regex = r'^(http:\/\/|https:\/\/|\/\/)[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$'
+
+# Regex for few of the widely used video hosting services
+player_regexes = {
+    'youtube': r'^(?:(?:https?:)?\/\/)?(?:www\.)?(?:youtu\.be\/|youtube(-nocookie)?\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((?:\w|-){11})(?:\S+)?$',
+    'vimeo': r'\/\/(player.)?vimeo.com\/([a-z]*\/)*([0-9]{6,11})[?]?.*',
+    'dailymotion': r'(?:.+dailymotion.com\/(video|hub|embed\/video|embed)|dai\.ly)\/([^_?]+)[^#]*(#video=([^_&]+))?',
+    'instagram': r'(?:(.*)instagram.com|instagr\.am)\/p\/(.[a-zA-Z0-9-_\.]*)',
+    'youku': r'(.*).youku\.com\/(v_show\/id_|embed\/)(.+)',
+}
+
+
+def get_video_source_data(video_url):
+    ''' Computes the valid source, document ID and regex match from given URL
+        (or False in case of invalid URL).
+    '''
+    if not video_url:
+        return False
+
+    if re.search(valid_url_regex, video_url):
+        youtube_match = re.search(player_regexes['youtube'], video_url)
+        if youtube_match and len(youtube_match.groups()[1]) == 11:
+            return ('youtube', youtube_match.groups()[1], youtube_match)
+        vimeo_match = re.search(player_regexes['vimeo'], video_url)
+        if vimeo_match:
+            return ('vimeo', vimeo_match.groups()[2], vimeo_match)
+        dailymotion_match = re.search(player_regexes['dailymotion'], video_url)
+        if dailymotion_match:
+            return ('dailymotion', dailymotion_match.groups()[1], dailymotion_match)
+        instagram_match = re.search(player_regexes['instagram'], video_url)
+        if instagram_match:
+            return ('instagram', instagram_match.groups()[1], instagram_match)
+        youku_match = re.search(player_regexes['youku'], video_url)
+        if youku_match:
+            youku_link = youku_match.groups()[2]
+            if '.html?' in youku_link:
+                youku_link = youku_link.split('.html?')[0]
+            return ('youku', youku_link, youku_match)
+    return False
+
+
+def get_video_url_data(video_url, **kwargs):
+    ''' Computes the platform name and embed_url from given URL
+        (or error message in case of invalid URL).
+    '''
+    source = get_video_source_data(video_url)
+    if not source:
+        return {'error': True, 'message': _('The provided url is invalid')}
+
+    autoplay = kwargs.get('autoplay') and '?autoplay=1&mute=1' or '?autoplay=0'
+    controls = kwargs.get('hide_controls') and '&controls=0' or ''
+    loop = kwargs.get('loop') and '&loop=1' or ''
+
+    # We directly use the provided URL as it is
+    embed_url = video_url
+    platform = source[0]
+    platform_id = source[1]
+    platform_match = source[2]
+    if platform == 'youtube' and len(platform_id) == 11:
```

Comment:
I did not look at that code too much as it was only moved but this indeed seems useless. So I removed it :+1:  I found the origin of this (in js): introduced 7 years ago by CHM when creating the web_editor app so I guess there was no real reason behind it.

I touched the regex but I kept the forced 11 in there: if you reach a youtube with >11 characters, it redirects to the 11 character one (so it is safe to **accept** more characters) but the regex still **needs** to **capture** only the first 11 otherwise youtube embed iframe and/or thumbnails are actually not working for some reason :shrug: